project/.pre-commit-config.yaml.j2

```diff
@@ -2,7 +2,7 @@
 minimum_pre_commit_version: 2.4.0
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.4.0
+    rev: {{ "v4.4.0" if python_requires < (3, 8) else "v4.6.0" }}
```

Comment:
**question (blocking):** is this check worthwhile considering everything prior to 3.7 is EOL?

Reply:
Good question. I didn't want to break 3.7 compatibility in this specific PR since it seemed unrelated to fixing the workflow. Would you prefer I add it to this one?

Reply:
Dropped 3.7 support now.

I noticed there is an incompatibility between black >=24 and asottile/reorder-python-imports which cannot be resolved and thus had to replace the latter with isort. Please see the commit message for details.